app/actions/nodeStorageActions.js

```diff
@@ -81,14 +108,20 @@ export const getRPCEndpoint = async (
   }
 }
 
-export const getNode = async (net: string): Promise<string> => {
+export const getNode = async (net: Net): Promise<string> => {
   const storage = await getStorage(`${STORAGE_KEY}-${net}`).catch(console.error)
-  if (!storage) return ''
-  return isEmpty(storage) ? '' : storage
+  const nodeInStorage = get(storage, 'node')
+  const expiration = get(storage, 'timestamp')
+  if (!nodeInStorage) return ''
+  if (!expiration) return ''
+  if (determineIfCacheIsExpired(expiration)) {
+    return ''
```

Comment:
these expressions can be combined into one `if (! nodeInStorage || !expiration || !determineIfCacheIsExpired(expiration)) { return '' }`